stream-chat-android-client/src/main/java/io/getstream/chat/android/client/parser/FilterObjectTypeAdapter.kt

```diff
@@ -34,19 +36,38 @@ internal class FilterObjectTypeAdapter(val gson: Gson) : TypeAdapter<FilterObjec
 }
 
 private fun FilterObject.toMap(): Map<*, *> = when (this) {
-    is AndFilterObject -> mapOf("\$and" to this.filterObjects.map(FilterObject::toMap))
-    is OrFilterObject -> mapOf("\$or" to this.filterObjects.map(FilterObject::toMap))
-    is NorFilterObject -> mapOf("\$nor" to this.filterObjects.map(FilterObject::toMap))
-    is ExistsFilterObject -> mapOf(this.fieldName to mapOf("\$exists" to true))
-    is NonExistsFilterObject -> mapOf(this.fieldName to mapOf("\$exists" to false))
+    is AndFilterObject -> mapOf(KEY_AND to this.filterObjects.map(FilterObject::toMap))
+    is OrFilterObject -> mapOf(KEY_OR to this.filterObjects.map(FilterObject::toMap))
+    is NorFilterObject -> mapOf(KEY_NOR to this.filterObjects.map(FilterObject::toMap))
+    is ExistsFilterObject -> mapOf(this.fieldName to mapOf(KEY_EXIST to true))
+    is NonExistsFilterObject -> mapOf(this.fieldName to mapOf(KEY_EXIST to false))
     is EqualsFilterObject -> mapOf(this.fieldName to this.value)
-    is NotEqualsFilterObject -> mapOf(this.fieldName to mapOf("\$ne" to this.value))
-    is ContainsFilterObject -> mapOf(this.fieldName to mapOf("\$contains" to this.value))
-    is GreaterThanFilterObject -> mapOf(this.fieldName to mapOf("\$gt" to this.value))
-    is GreaterThanOrEqualsFilterObject -> mapOf(this.fieldName to mapOf("\$gte" to this.value))
-    is LessThanFilterObject -> mapOf(this.fieldName to mapOf("\$lt" to this.value))
-    is LessThanOrEqualsFilterObject -> mapOf(this.fieldName to mapOf("\$lte" to this.value))
-    is InFilterObject -> mapOf(this.fieldName to mapOf("\$in" to this.values))
-    is NotInFilterObject -> mapOf(this.fieldName to mapOf("\$nin" to this.values))
-    is AutocompleteFilterObject -> mapOf(this.fieldName to mapOf("\$autocomplete" to this.value))
+    is NotEqualsFilterObject -> mapOf(this.fieldName to mapOf(KEY_NOT_EQUALS to this.value))
+    is ContainsFilterObject -> mapOf(this.fieldName to mapOf(KEY_CONTAINS to this.value))
+    is GreaterThanFilterObject -> mapOf(this.fieldName to mapOf(KEY_GREATER_THAN to this.value))
+    is GreaterThanOrEqualsFilterObject -> mapOf(this.fieldName to mapOf(KEY_GREATER_THAN_OR_EQUALS to this.value))
+    is LessThanFilterObject -> mapOf(this.fieldName to mapOf(KEY_LESS_THAN to this.value))
+    is LessThanOrEqualsFilterObject -> mapOf(this.fieldName to mapOf(KEY_LESS_THAN_OR_EQUALS to this.value))
+    is InFilterObject -> mapOf(this.fieldName to mapOf(KEY_IN to this.values))
+    is NotInFilterObject -> mapOf(this.fieldName to mapOf(KEY_NOT_IN to this.values))
+    is AutocompleteFilterObject -> mapOf(this.fieldName to mapOf(KEY_AUTOCOMPLETE to this.value))
+    is DistinctFilterObject -> mapOf(KEY_DISTINCT to true, KEY_MEMBERS to this.memberIds)
```

Comment:
Feels really weird that this distinct filter always operates on members, but it doesn't have that in its name. The API looks generic from the outside (maybe usable for channel filtering), but then it's something very specific internally.